src/Dialogs/SyncWarningDialog.vala

```diff
@@ -26,44 +26,54 @@
  * Authored by: Scott Ringwelski <sgringwe@mtu.edu>
  */
 
-public class Noise.SyncWarningDialog : Gtk.Window {
+public class Noise.SyncWarningDialog : Gtk.Dialog {
+    private enum ResponseId {
+        IMPORT_MEDIA = 1,
+        CONTINUE,
+        STOP
+    }
+
     Device d;
     Gee.TreeSet<Media> to_sync = new Gee.TreeSet<Media> ();
     Gee.TreeSet<Media> to_remove = new Gee.TreeSet<Media> ();
-    
-    private Gtk.Box content;
-    private Gtk.Box padding;
-    
-    Gtk.Button importMedias;
-    Gtk.Button sync;
-    Gtk.Button cancel;
-    
-    public SyncWarningDialog(Device d, Gee.Collection<Media> to_sync, Gee.Collection<Media> removed) {
+
+    Gtk.Widget import_media_button;
+    Gtk.Widget continue_button;
+    Gtk.Widget stop_button;
+
+    construct {
+        type_hint = Gdk.WindowTypeHint.DIALOG;
+        modal = true;
+        transient_for = App.main_window;
+        destroy_with_parent = true;
+
+        set_default_size (475, -1);
+        resizable = false;
+        border_width = 20;
+
+        // create buttons
+        import_media_button = add_button (_("Import media to Library"), ResponseId.IMPORT_MEDIA);
+        continue_button = add_button (_("Continue Syncing"), ResponseId.CONTINUE);
+        stop_button = add_button (_("Stop Syncing"), ResponseId.STOP);
+        response.connect (on_response);
+
+        import_media_button.sensitive = !libraries_manager.local_library.doing_file_operations ();
+        continue_button.sensitive = !libraries_manager.local_library.doing_file_operations ();
+    }
+
+    public SyncWarningDialog (Device d, Gee.Collection<Media> to_sync, Gee.Collection<Media> removed) {
         this.d = d;
         this.to_sync.add_all (to_sync);
         this.to_remove.add_all (removed);
 
-        // set the size based on saved gconf settings
-        //this.window_position = WindowPosition.CENTER;
-        this.type_hint = Gdk.WindowTypeHint.DIALOG;
-        this.set_modal(true);
-        this.set_transient_for(App.main_window);
-        this.destroy_with_parent = true;
-        
-        set_default_size(475, -1);
-        resizable = false;
-        
-        content = new Gtk.Box(Gtk.Orientation.VERTICAL, 10);
-        padding = new Gtk.Box(Gtk.Orientation.HORIZONTAL, 20);
-        
+        Gtk.Box content = get_content_area ();
```

Comment:
This and most of the following code in the constructor can be transferred to the construct block.   Only retain in the constructor code that requires reference, direct or indirect, to parameters "to_sync" and "remove".